src/main/java/fi/otavanopisto/kuntaapi/server/images/ScaledImageStore.java

```diff
@@ -0,0 +1,138 @@
+package fi.otavanopisto.kuntaapi.server.images;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.smile.SmileFactory;
+
+import fi.otavanopisto.kuntaapi.server.id.AttachmentId;
+import fi.otavanopisto.kuntaapi.server.integrations.AttachmentData;
+
+/**
+ * Image store for scaled images
+ * 
+ * @author Antti Leppä
+ */
+@ApplicationScoped
+public class ScaledImageStore {
+ 
+  @Inject
+  private Logger logger;
+  
+  /**
+   * Returns scaled image data
+   * 
+   * @param kuntaApiAttachmentId attachmentId
+   * @param size size
+   * @return scaled image data or null if not found
+   */
+  public AttachmentData getStoredImage(AttachmentId kuntaApiAttachmentId, Integer size) {
+    File imageFile = getImageFile(kuntaApiAttachmentId, size);
+    if (imageFile.exists()) {
+      ObjectMapper objectMapper = getObjectMapper();
+      try {
+        return objectMapper.readValue(imageFile, AttachmentData.class);
+      } catch (IOException e) {
+        logger.log(Level.WARNING, String.format("Failed to read stored scaled image %s", imageFile.getName()), e);
+        if (!imageFile.delete()) {
+          logger.log(Level.WARNING, String.format("Failed to delete stored scaled image %s", imageFile.getName()), e);  
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Failed to delete stored scaled image %s" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)